### ISSUE TITLE
MonadError#adaptError could be on ApplicativeError instead

### ISSUE THREAD

**Issue by bplommer** [typelevel/cats#2685]
`MonadError#adaptError` is defined as 
```scala
def adaptError[A](fa: F[A])(pf: PartialFunction[E, E]): F[A] =
  flatMap(attempt(fa))(_.fold(e => raiseError(pf.applyOrElse[E, E](e, _ => e)), pure))
```
But it could be defined on `ApplicativeError` simply as
```scala
def adaptError[A](fa: F[A])(pf: PartialFunction[E, E]): F[A] =
  recoverWith(fa)(pf.andThen(raiseError))
```
I haven't opened a PR because this would be a breaking change.

**Comment by bplommer** [comment]
@kailuowang I'm happy to pick this up. I it will need a different name in the `ApplicativeError` syntax to prevent ambiguity - maybe `applicativeAdaptError`?

**Comment by kailuowang** [comment]
@bplommer thanks, yeah, that's a good catch. `applicativeAdaptError` sounds too long. how about `adaptErr` or `modifyError` with a comment on why it can't be the same as the one in MonadError for now? we shall be able to change this in 2.0.

**Comment by kubukoz** [comment]
Can this be closed? :)

**Comment by kailuowang** [comment]
I still like to move this method in the type class in 2.1

**Comment by kubukoz** [comment]
On the >2.11 branch, right?

**Comment by kailuowang** [comment]
Right, which is master.

**Comment by travisbrown** [comment]
For what it's worth it should have been possible to fix the syntax in 2.0 by making the `MonadErrorOps` `adaptError` package private, but I just tried it for 2.1 and ran into a compiler bug: https://github.com/scala/bug/issues/11787

(`private` or `private[MonadErrorOps]` avoid the bug but break bincompat.)
